proctoru/proctoru.py

```
@@ -373,7 +373,8 @@ def student_view(self, context=None):
 
                     allowed_keep_old_exam = False
 
-                    if exam_start_date_time <= old_exam_time <= exam_end_date_time:
+                    if old_exam_time.tzinfo is not None  \
+                            and (exam_start_date_time <= old_exam_time <= exam_end_date_time):
```

This line is too complicated. datetime comparison should always be simple to read for human.

I simply do not understand this statement whithout reading Python doc:

    exam_start_date_time <= old_exam_time <= exam_end_date_time

I propose to split it in two, please verify this is equivalent:

    if old_exam_time.tzinfo and \
        exam_start_date_time <= old_exam_time and \
        exam_start_date_time <= exam_end_date_time:

---

Ok. Seems better. Cheers.
A quick beginner question: are we sure that the last two comparison will not be executed if the first condition fails (old_exam_time.tzinfo) ? I would not want to raise an exception when comparing a date ith old_exam_time.
Also, you are saying that old_exam_time.tzinfo  == old_exam_time.tzinfo is Not None right ?

I will do the ammends later this week and test this carefully.

---

Python will stop evaluating when first condition is False when using `and` operator